### PR TITLE
`azurerm_nginx_deployment`: Deprecate `logging_storage_account` block

### PR DESCRIPTION
**Pull request by puneetsarna**
NGINXaaS for Azure is deprecating the use of
`logging_storage_account` block in favor of using
Azure native logging configured via Azure
Diagnostic Settings.

## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [ ] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [ ] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [ ] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [ ] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [ ] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Comment by puneetsarna** [comment]
Marking this as a draft pending an internal discussion with our team.

File: internal/services/nginx/nginx_deployment_data_source.go [hunk 1]
```diff
@@ -249,12 +250,14 @@ func (m DeploymentDataSource) Read() sdk.ResourceFunc {
 					output.NginxVersion = pointer.ToString(props.NginxVersion)
 					output.DiagnoseSupportEnabled = pointer.ToBool(props.EnableDiagnosticsSupport)
 
-					if props.Logging != nil && props.Logging.StorageAccount != nil {
-						output.LoggingStorageAccount = []LoggingStorageAccount{
-							{
-								Name:          pointer.ToString(props.Logging.StorageAccount.AccountName),
-								ContainerName: pointer.ToString(props.Logging.StorageAccount.ContainerName),
-							},
+					if !features.FivePointOhBeta() {
```
Comment (stephybun) [inline internal/services/nginx/nginx_deployment_data_source.go:253]: The `logging_storage_account` property in the data source schema needs to be removed conditionally using the `features.FivePointOhBeta` flag as well

File: internal/services/nginx/nginx_deployment_resource.go [hunk 2]
```diff
@@ -264,6 +264,27 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 		"tags": commonschema.Tags(),
 	}
 
+	if !features.FivePointOhBeta() {
+		resource["logging_storage_account"] = &pluginsdk.Schema{
+			Deprecated: "The `logging_storage_account` block has been deprecated and will be removed in v5.0 of the AzureRM Provider. To enable logs, use `azurerm_monitor_diagnostic_setting` resource instead.",
```
Comment (stephybun) [inline internal/services/nginx/nginx_deployment_resource.go:269]:
```suggestion
			Deprecated: "The `logging_storage_account` block has been deprecated and will be removed in v5.0 of the AzureRM Provider. To enable logs, use the `azurerm_monitor_diagnostic_setting` resource instead.",
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @puneetsarna! Aside from the one comment left in-line this look good! Once that comment is resolved this should be good to go.

File: internal/services/nginx/nginx_deployment_data_source.go (hunk 1 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_deployment_data_source.go:253]: 👍

File: internal/services/nginx/nginx_deployment_resource.go (hunk 2 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_deployment_resource.go:269]: 👍

Comment (puneetsarna) [reply internal/services/nginx/nginx_deployment_resource.go]: done.

File: internal/services/nginx/nginx_deployment_data_source.go (hunk 1 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_deployment_data_source.go:257]: done.

**Comment by puneetsarna** [comment]
@stephybun Thanks for reviewing this. I have pushed fixes to the PR around properly deprecating the block.

```shell
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/nginx -run=TestAccNginxDeploymentDataSource_basic -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccNginxDeploymentDataSource_basic
=== PAUSE TestAccNginxDeploymentDataSource_basic
=== CONT  TestAccNginxDeploymentDataSource_basic
--- PASS: TestAccNginxDeploymentDataSource_basic (409.10s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx	414.838s
(reverse-i-search)`make': make e2e.nginx args="-k test_insecure_upstreams"
terraform-provider-azurerm $ make acctests SERVICE='nginx' TESTARGS='-run=TestAccNginxDeployment_basic'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/nginx -run=TestAccNginxDeployment_basic -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccNginxDeployment_basic
=== PAUSE TestAccNginxDeployment_basic
=== CONT  TestAccNginxDeployment_basic
--- PASS: TestAccNginxDeployment_basic (395.95s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx	398.634s
```

**Review by stephybun — APPROVED** [review]
Thanks @puneetsarna! I totally missed that the upgrade guide wasn't updated. Hope you don't mind I pushed a commit to update the upgrade guide to your branch.

LGTM 💯

**Review by catriona-m — APPROVED** [review]
LGTM!

**Comment by puneetsarna** [comment]
@stephybun Thanks for taking care of the guide 👍.

**Pull request merged**